facets.wdl

```diff
@@ -124,41 +127,51 @@ task Pileup {
 task FACETS {
     String pair_name
     File pileup
+    Int ndepth
     Int cval
     Int maxiter
+    Int seed_initial
+    Int seed_iterations
 
-    Int? memoryGB
-    Int? diskGB
+    Int? preemptible_attempts
+    Int? memoryGB = 4
+    Int? diskGB = ceil(1.1 * (size(pileup, "G"))) + 20
 
     command <<<
-        Rscript /facets.R ${pair_name} ${pileup} ${cval} ${maxiter}
+        Rscript /facets.R ${pair_name} ${pileup} ${ndepth} ${cval} ${maxiter} ${seed_initial} ${seed_iterations}
     >>>
 
     runtime {
-        docker: "vanallenlab/facets:v0.5.14"
+        docker: "vanallenlab/facets:v0.5.14.1"
         memory: "${memoryGB} GB"
         disks: "local-disk ${diskGB} SSD"
+        preemptible: preemptible_attempts
     }
 
     output {
         File genome_segments = "${pair_name}.genome_segments.pdf"
         File diagnostics_plot = "${pair_name}.diagnostic_plot.pdf"
+        File iterations_plot = "${pair_name}.facets_iterations.pdf"
         File cncf = "${pair_name}.facets_cncf.txt"
         File summary = "${pair_name}.facets_output.txt"
         File flags = "${pair_name}.facets_flags.txt"
         File emflags = "${pair_name}.facets_emflags.txt"
+        File iterations = "${pair_name}.facets_iterations.txt"
         String purity = read_string("purity.txt")
         String ploidy = read_string("ploidy.txt")
         String log_likelihood = read_string("log_likelihood.txt")
         String dip_log_r = read_string("dip_log_r.txt")
+        String seed_used = read_string("seed_used.txt")
+        String n_iterations_na_purity = read_string("number_iterations_with_na_purity.txt")
     }
 }
 
 task InferWGD {
     File cncf
 
-    Int? memoryGB
-    Int? diskGB
+    Int? preemptible_attempts
+    Int? memoryGB = 4
+    Int? diskGB = ceil(1.1 * (size(cncf, "G"))) + 20
```

Comment:
Not important, but I hope that this particular python script has no need for the 20 GB constant